static/app/components/events/highlights/highlightsDataSection.tsx

```diff
@@ -1,20 +1,217 @@
+  const highlightTags = detailedProject?.highlightTags ?? [];
+
+  // The API will return default values for tags/context. The only way to have none is to set it to
+  // empty yourself, meaning the user has disabled highlights
+  const hasDisabledHighlights =
+    Object.values(highlightContext).flat().length === 0 && highlightTags.length === 0;
+
+  const viewAllButton = viewAllRef ? (
+    <Button
+      onClick={() => viewAllRef?.current?.scrollIntoView({behavior: 'smooth'})}
+      size="xs"
+    >
+      {t('View All')}
+    </Button>
+  ) : null;
+
+  const highlightContextDataItems = getHighlightContextData({
+    event,
+    project,
+    organization,
+    highlightContext,
+  });
+  const highlightContextRows = highlightContextDataItems.reduce<React.ReactNode[]>(
+    (rowList, {alias, data}, i) => {
+      const meta = getContextMeta(event, alias);
+      const newRows = data.map((item, j) => (
+        <HighlightContextContent
+          key={`highlight-ctx-${i}-${j}`}
+          meta={meta}
+          item={item}
+          alias={alias}
+          config={{includeAliasInSubject: true}}
+        />
+      ));
+      return [...rowList, ...newRows];
+    },
+    []
+  );
+
+  const highlightTagItems = getHighlightTagData({event, highlightTags});
+  const highlightTagRows = highlightTagItems.map((content, i) => (
+    <EventTagsTreeRow
+      key={`highlight-tag-${i}`}
+      content={content}
+      event={event}
+      tagKey={content.originalTag.key}
+      projectSlug={project.slug}
+    />
+  ));
+
+  const rows = [...highlightTagRows, ...highlightContextRows];
+  const columns: React.ReactNode[] = [];
+  const columnSize = Math.ceil(rows.length / columnCount);
+  for (let i = 0; i < rows.length; i += columnSize) {
+    columns.push(
+      <HighlightColumn key={`highlight-column-${i}`}>
+        {rows.slice(i, i + columnSize)}
+      </HighlightColumn>
+    );
+  }
+
+  function openEditHighlightsModal() {
+    openModal(
+      deps => (
+        <EditHighlightsModal
+          event={event}
+          highlightContext={highlightContext}
+          highlightTags={highlightTags}
+          project={detailedProject ?? project}
+          highlightPreset={detailedProject?.highlightPreset}
+          {...deps}
+        />
+      ),
+      {modalCss: highlightModalCss, onClose: refetch}
+    );
+  }
+
+  return (
+    <EventDataSection
+      title={t('Event Highlights')}
+      data-test-id="event-highlights"
+      type="event-highlights"
+      actions={
+        <ButtonBar gap={1}>
+          {viewAllButton}
+          <Button size="xs" icon={<IconEdit />} onClick={openEditHighlightsModal}>
+            {t('Edit')}
+          </Button>
+        </ButtonBar>
+      }
+    >
+      <HighlightContainer columnCount={columnCount} ref={containerRef}>
+        {isLoading ? (
+          <EmptyHighlights>
+            <HighlightsLoadingIndicator hideMessage size={50} />
```

Comment:
One thing I don't _love_ about this is that on first pageload the height will change when this has finished loading, which causes important things like the stacktrace to get shifted down. Not a _huge_ deal, but good to avoid if we can. Is this project available in the ProjectsStore by chance? Or can we start the project request before the event has finished loading?

Reply:
I will record this as feedback and notion and try and work out a way around this, I need to make a detailed project query to get the highlights, so height jumps around. If I can find a way to make that hook call higher in the tree and pass it down, that'd probably alleviate this. Thanks for mentioning it though!

Reply:
A couple things that might make it easier for you:

- Could simply call `useDetailedProject()` somewhere higher in the hierarchy and it will be available in the cache later on, no need to pass it down in props
- If it's available in the ProjectsStore already, you could send that in with the `initialData` prop of useApiQuery to seed it with the data that's already available